### PR TITLE
Attribute Sorting/Filtering/Graphing

### PR DESCRIPTION
**Pull request by BryonLewis**
Attributes Filter/Sorting:

New Features:
- Fixed import of JSON files so that Attributes are calculated when a user imports a trackJSON file.  Before they were only calculated when a VIAME-CSV was imported.
- Attributes Subpanel CSS updated to fix minor issues from displaying frame numbers to not displaying Track Attributes if too many Detection Attributes Existed.
- Added a Sort option for Attributes which will sort Alphabetically by key or Numerically (placing non-numeric values at the bottom).
- Attributes now have a color associated with them automatically at run time.  I am not saving this value in the Cofiguration.json file.  This could be a further update if we allow the user to choose attribute color in the future.
- Filtering Icon for attributes is active when Attribute filtering is turned on.
- Attribute Details Panel
    - Attribute Filtering
        - Allows filtering the display of Attributes based on Numeric/Text/Boolean values or if a specific key is visible.
        - Can filter Track and Detection attributes independently
        - Filters are applied in the order they are displayed to allow cascading filters.
    - Timeline Visualization
        - Shows a graph representing the numeric values of attributes for the selected track
        - Limited currently to numeric values only but will updated in the future to include text/boolean values.
        - The graph icon is displayed next to the "Detections" and "Events" button.
        - Graphing is based on using a "Key" filter where it will only graph the accepted keys.
 -  Extras:
     - Hid the Groups Button for the timeline if no groups exist.

Dev Information:

- **useAttributes** - This file has expanded largely to include the AttributeFilter type definitions and storage of the data to apply filters.  If this goes further there may be a need to add more data to it.
   - Storage for the filters, timelines, and functions for modifying the filters/timelines 
- **provides.ts** - I included a separate `useAttributeFilters` to the provies.ts to use the specific values associated with AttributesFiltering and visualization.
- Vue Components
    - There are a few Vue components that were needed to properly display information in the Attribute Details Panel.  These are mostly the filtering types and an ability to edit the filtering settings.
- **Attributes Timeline**
    - Some small modifications to the LineChart.vue to create a linear graph instead of step and also allow for highlighting of hovered lines.

### CODE REVIEW

File: client/dive-common/components/AttributesSubsection.vue [hunk 1]
```diff
@@ -39,8 +42,12 @@ export default defineComponent({
     const readOnlyMode = useReadOnlyMode();
     const { frame: frameRef } = useTime();
     const selectedTrackIdRef = useSelectedTrackId();
+    const { attributeFilters, sortAndFilterAttributes } = useAttributesFilters();
     const cameraStore = useCameraStore();
     const activeSettings = ref(true);
+    const sortingMethods = ['a-z', '1-0'];
+    const sortingMethodIcons = ['mdi-sort-alphabetical-ascending', 'mdi-sort-numeric-ascending'];
```
[the remark this answers is not in the document]
Comment (BryonLewis) [reply client/dive-common/components/AttributesSubsection.vue:49]: I copied the behavior in the type list.  It has the same sort of sorting.  Alphabetically or Numerically by count.  I wanted to stick with the same paradigm for this.

File: docs/UI-AttributeDetails.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Attribute filtering will filter the values for selected type to reduce the number of them displayed.  The filters can be applied to 
+"Track" or "Detection" attributes and are applied in order.
+
+Attribute filters have an "Applies To" list which is a list of the Attribute key names that the filter will apply to.
+There is a special name called 'all' which will cause the filter to apply to all items.
+The settings for individual filters can be accessed by clicking on the ==:material-cog:==
+
+### Numeric Filters
+
+Numeric filters will only operate on the filters that have numeric values.
```
[the remark this answers is not in the document]
Comment (BryonLewis) [reply docs/UI-AttributeDetails.md:18]: "only operates on attributes with numeric values". 

Similar for each section.

**Comment by BryonLewis** [comment]
Change "Key Filter" for the timeline to "Timeline Filter".

**Comment by BryonLewis** [comment]
Change "Filter Type" to some terminology which isn't already used.
Maybe Switch to using radio buttons for the track/detection to swap between the filters.

File: client/dive-common/components/AttributesSubsection.vue [hunk 3]
```diff
@@ -205,7 +210,7 @@ export default defineComponent({
               <v-icon small>
                 mdi-plus
               </v-icon>
-              Attribute
+              Add
```
Comment (marySalvi) [inline client/dive-common/components/AttributesSubsection.vue:213]: image
I had something more like this in mind for the attribute headers. What do you think?

File: client/src/components/AttributeFilters.vue [hunk 4]
```diff
@@ -118,8 +134,19 @@ export default defineComponent({
           },
         };
       }
-      addAttributeFilter(0, currentTab.value, newFilter);
-      addFilterDialog.value = false;
+      addFilterFilter.value = newFilter;
+      addFilterPosition.value = 2;
+      //addAttributeFilter(0, currentTab.value, newFilter);
+      //addFilterDialog.value = false;
```
Comment (marySalvi) [inline client/src/components/AttributeFilters.vue:140]: can these comments be removed?

**Review by marySalvi — COMMENTED** [review]
This looks great! 

Couple of thoughts that I don't know if they should be for this PR or another: 
I think that when you click on the filter button for either track or detection, the radio button selected should reflect whichever you clicked to open it.

Also I think 'Draw Timeline' should trigger the timeline to switch to the attributes linechart

**Review by marySalvi — APPROVED** [review]
This looks awesome! :shipit:

**Pull request merged**